controller/workitem.go

```diff
@@ -111,18 +136,44 @@ func (c *WorkitemController) Update(ctx *app.UpdateWorkitemContext) error {
 	if !authorized {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to access the space"))
 	}
+
+	if ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.BaseType != nil &&
+		ctx.Payload.Data.Relationships.BaseType.Data.ID != wi.Type {
+		// Store new values of type and version
+		newTypeID := ctx.Payload.Data.Relationships.BaseType
+		newVersion := ctx.Payload.Data.Attributes[workitem.SystemVersion]
+
+		// Remove version and base type from payload
+		delete(ctx.Payload.Data.Attributes, workitem.SystemVersion)
+		ctx.Payload.Data.Relationships.BaseType = nil
+
+		// Ensure we do not have any other change in payload except type change
```

Comment:
@jarifibrahim Why do we ensure that we do not have any other change in payload except type change?

Reply:
@kwk 
>One for the work item type change itself and then another one for the made changes to the work item fields.

When a workitem is changed from `Type A` to `Type B`.
In state 1 - Workitem revision has fields of type A and actual type is A
In state 2 - Workitem revision has fields of type A and actual type is B
In state 3 - Workitem revision has fields of type B and actual type is B
Even in this way of saving `2 revisions` we will have an inconsistent state. State 2 has different fields as compared to its actual type. The `Event` system has to deal with state 2 even if we store 2 revisions.

Reply:
@kwk @jarifibrahim if the workitem update payload includes update to field + update to workitem type, then it could be handled in two parts:

1. Update the fields first
2. Update the WIT

I am not really sure about imposing a restriction that field update cannot happen with WIT update.

Reply:
Based on the discussion with @DhritiShikhar  and Nimisha, I've created a separate task for this https://openshift.io/openshiftio/Openshift_io/plan/detail/433